### ISSUE TITLE
ABAP Code Review Guide - abapGit Organization Configuration

### ISSUE THREAD

**Issue by OberstVonGatow** [SAP/styleguides#201]
I'm searching for a way to implement Code Reviews for our Organization. We are 3-4 Inhouse Developers and X external Developers.

- Is it possible to (and should we) have every Custom Code Package in abapGit?
We often don't have a specific big project. Often small changes (maybe only one UserExit) or maybe already done projects with minor changes from time to time. Could we take one really big ZALL package and put all other custom packages in it and on git?
Would it be a performance issue? Or an Issue with abaplint?

- We don't use "Transports of Copies". Rather we use lots of single TR (We think this keeps the System "clean". You always see which version/TR is in which System including Q-System)
I think about changing "Check PR is released, if not the request cannot be released" to "... the request cannot be imported" with a different Badi/Userexit before importing in Production System.

Thanks for the Input and the nice guide 👍

**Comment by larshp** [comment]
### One big package
yes, its possible, but also consider that notifications on github are scoped to a repository, so if different parts of your organization have different responsibilities, it might be beneficial to organize the repos like the organization

### Performance
try 😄 Install the abapGit developer edition, run "package to zip", that way you can see how long it takes for abapGit to serialize, make sure the serialization runs in parallel
![image](https://user-images.githubusercontent.com/5888506/115820736-c0b5e200-a401-11eb-8d68-6e0f05e6d31f.png)
When you have the zip file, extract it to local pc, "npm install @abaplint/cli -g" and run "time abaplint" 

### ToC
Yea, https://github.com/abapGit/abapgit-review-example is just an example to get started, it can be adjusted according to your work processes. For each released TR you can amend commits to the branch, just add some logic to determine the branch name for each transport request

**Comment by xtough** [comment]
...wouldn't one big git repo also have the penalty of having to clone that ZALL package to every dev system where you want to make use of any of its contents? Or is there some sort of "sparse checkout" feature of abapGit that I am not aware of?

**Comment by larshp** [comment]
I assume there is one central dev system in the landscape, and abapGit will only be used to export code so that code reviews can be done

**Comment by OberstVonGatow** [comment]
First of all thank you both for your quick answers.
Lars is right sorry that i missed to mention that. We have one central dev system and abapGit/Git would "only" be there for code reviews. External Devs also connect to the central dev system now and in the near future.

Great that "One big package/repo" is possible. I think we will build a prototype and check if one big repo (ZALL) or one repo per module  (Z_MM,Z_SD) is better for us.

I would also prefer one repo for each module but i think quite often we would have one rfc affecting multiple modules.
If that would happen and the Dev is not aware of it we would get problems.
Because only one repo per SAP Transportrequest is possible if i'm right.

**Comment by xtough** [comment]
@OberstVonGatow can we close this issue? Or is there anything your would like to add to the code review guide?
